weave/artifact_wandb.py

```diff
@@ -489,7 +499,10 @@ def _set_read_artifact_uri(self, uri):
     def _saved_artifact(self):
         if self._read_artifact is None:
             uri = self._read_artifact_uri
-            path = f"{uri.entity_name}/{uri.project_name}/{uri.name}:{uri.version}"
+            if isinstance(uri, WeaveWBArtifactByIDURI):
+                path = f"{uri.path_root}/{uri.artifact_id}/{uri.name}:{uri.version}"
```

Comment:
There lines:
```
path = f"{uri.path_root}/{uri.artifact_id}/{uri.name}:{uri.version}"
path = f"{uri.entity_name}/{uri.project_name}/{uri.name}:{uri.version}"
uri_path = f"{uri_parts.entity_name}/{uri_parts.project_name}/{uri_parts.name}:{uri_parts.version}"
uri_path = f"{uri_by_id_parts.path_root}/{uri_by_id_parts.artifact_id}/{uri_by_id_parts.name}:{uri_by_id_parts.version}"
```

Seem like they should be defined as methods off the object, as opposed to manually constructed here.